examples/systemcalls/systemcalls.c

```diff
@@ -83,17 +94,46 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
-
+    va_end(args);
 
 /*
- * TODO
  *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
  *   redirect standard out to a file specified by outputfile.
  *   The rest of the behaviour is same as do_exec()
- *   
 */
+	pid_t pid = fork();
+	if (pid == -1) { // failure
+		perror("fork");
+		return false;
+	} else if (pid == 0) { // child
+		int fd = open(outputfile, O_WRONLY);
+		if (fd < 0) {
+			perror("open");
+			abort();
+		}
 
-    va_end(args);
-    
-    return true;
+		int nfd = dup2(fd, fileno(stdout));
```

Comment:
@georgehodgkins consider how you could combine with the function above and avoid duplicating logic.